integration_tests/src/main/python/string_test.py

```diff
@@ -779,4 +779,20 @@ def test_like_complex_escape():
                 'a like "\\%SystemDrive\\%\\\\\\\\Users%"',
                 'a like "_oo"'),
             conf={'spark.sql.parser.escapedStringLiterals': 'true'})
- 
+
```

Comment:
You need a test with no nulls in it, because when I run that I get an error like.

```
Caused by: ai.rapids.cudf.CudfException: CUDF failure at: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-516-cuda11/thirdparty/cudf/cpp/src/transform/mask_to_bools.cu:42: nullmask is null
  at com.nvidia.spark.rapids.jni.CastStrings.toIntegersWithBase(Native Method)
  at com.nvidia.spark.rapids.jni.CastStrings.toIntegersWithBase(CastStrings.java:108)
  at org.apache.spark.sql.rapids.GpuConv.doColumnar(stringFunctions.scala:2071)
  at com.nvidia.spark.rapids.GpuTernaryExpression.$anonfun$columnarEval$8(GpuExpressions.scala:443)
```